.azure_pipelines/conda_envs/windows_2019_py_38_env.yml

```
@@ -0,0 +1,15 @@
+dependencies:
+  - python=3.8.1
+  - numpy
+  - ninja
+  - pyyaml
+  - mkl
+  - mkl-include
+  - setuptools
+  - cmake
+  - cffi
+  - typing_extensions
+  - future
+  - six
+  - requests
+  - dataclasses
```

Please correct me if I'm wrong, but those are just copy-n-paste from requirements.txt?
In that case, wouldn't it be better to do something like
```suggestion
  - pip:
    - -r requirements.txt
 ```

---

Yes, that is true. The env ymls only differ on the Python version as well, so it would be best to have one env yaml with dynamic Python versioning and with `pip: -r requirements.txt`.

---

To the best of my knowledge (I'm conda noob), but you can create the environment with selected python version and than use common .yaml file to install the rest of required packages, but doing something like the following:
```
 conda create -n foo python=3.8
 conda install -n foo -f common.yaml
```